test/terminate_when_main_terminates_test.ml

```diff
@@ -0,0 +1,14 @@
+[@@@warning "-8"]
+
+open Riot
+
+exception Error
+
+let main () =
+  (* spawn process that should throw which should terminate main process *)
+  let _pid = spawn_link (fun () -> raise Error) in
+  ()
```

Comment:
i think the test should set it up so that:

1. we spin up a new process that is linked
2. that process awaits a message with receive and immediately sends it back to the main process
3. then the main process receive a message with a timeout of say 5_000_000L 
5. if we find a message, we exit with an error
6. else AFTER the Riot.run call we should do a print that the test is Ok


wdyt?